go-controller/pkg/clustermanager/egressservice/egressservice_cluster_node.go

```diff
@@ -208,6 +209,16 @@ func (c *Controller) syncNode(key string) error {
 			}
 		}
 
+		labelsToRemove := map[string]any{}
```

Comment:
We are removing all of the egressservice labels for nodes that do not have a state in our cache, I am concerned that this could race on startup where we would remove all labels here for them to be added back in the esvc controller. 
Shouldn't we do it in the main controller instead?